services/src/main/java/org/apache/druid/cli/CliPeon.java

```diff
@@ -565,6 +553,28 @@ static void configureIntermediaryData(Binder binder)
     shuffleClientBiddy.addBinding("deepstore").to(DeepStorageShuffleClient.class).in(LazySingleton.class);
   }
 
+  static Map<String, Object> heartbeatDimensions(Task task)
+  {
+    ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
+    builder.put(DruidMetrics.TASK_ID, task.getId());
+    builder.put(DruidMetrics.DATASOURCE, task.getDataSource());
+    builder.put(DruidMetrics.TASK_TYPE, task.getType());
+    builder.put(DruidMetrics.GROUP_ID, task.getGroupId());
+    Map<String, Object> tags = task.getContextValue(DruidMetrics.TAGS);
+    if (tags != null && !tags.isEmpty()) {
+      builder.put(DruidMetrics.TAGS, tags);
+    }
+
+    if (task instanceof SeekableStreamIndexTask) {
+      SeekableStreamIndexTask streamingTask = (SeekableStreamIndexTask) task;
+      if (streamingTask.getCurrentRunnerStatus() != null) {
+        builder.put(DruidMetrics.STATUS, streamingTask.getCurrentRunnerStatus());
```

Comment:
This is not very likely to happen but the value of the `getCurrentRunnerStatus()` could change between these two calls. Probably better to assign it to a variable instead.

Reply:
good call, fixed